### DISCUSSION TITLE
OpenURL not compiling in an iOS app with watchOS target embedded

### DISCUSSION THREAD

**Discussion by trispo** [pointfreeco/swift-composable-architecture#1422] [Beta]
As mentioned in the title the newly added file [OpenURL](https://github.com/pointfreeco/swift-composable-architecture/blob/protocol-beta/Sources/Dependencies/Dependencies/OpenURL.swift) does not compile when the target iOS app contains a watchOS app. I think to fix this the file should contain a `#if !os(watchOS)` statement. Shall I open a PR for this?

**Comment by mbrandonw** [comment]
Hi @trispo, sorry about that!  We meant to push that to `protocol`, not `protocol-beta`, because we didn't have a chance to experiment with it on watchOS/tvOS. Please do feel free to open a PR if you have time.

**Reply by mbrandonw** [reply]
Interestingly CI doesn't have any test failures even though it does build the library for watchOS. And looking at the implementation there doesn't seem to be any non-watchOS symbols being used.

Can you share what the exact build failure is?

**Reply by trispo** [reply]
UIApplication does not exist on watchOS.

![image](https://user-images.githubusercontent.com/1045441/193037139-75117eb6-8945-40b0-b697-4c63bbf44be6.png)

**Reply by mbrandonw** [reply]
Ahhh, ok! Then we should definitely just surround the whole file with `#if !os(watchOS)` until we get more time to experiment with this feature.

**Reply by trispo** [reply]
With these changes it compiles:

![CleanShot 2022-09-29 at 14 57 28](https://user-images.githubusercontent.com/1045441/193037542-63a73c22-3d3c-404a-96a8-9e14024b9be8.png)

**Reply by trispo** [reply]
> Ahhh, ok! Then we should definitely just surround the whole file with `#if !os(watchOS)` until we get more time to experiment with this feature.

OK, will do
